scenarios/04_Battle_of_Barnon.cfg

```diff
@@ -1224,6 +1274,93 @@
         [/if]
     [/event]
 
+    [event]
+        name=enemies defeated
+        first_time_only=no # because this could fire when Hoyre is the only enemy, in which case we need to trigger the betrayal, which means it could happen again
+        [if]
+            [variable]
+                name=ano_barnon_tr
+                equals=no
+            [/variable]
+            [and]
+                [have_unit]
+                    id=Bor Cryne
+                    [filter_side]
+                        [allied_with]
+                            side=1
+                        [/allied_with]
+                    [/filter_side]
+                [/have_unit]
+            [/and]
+            [and]
+                [have_unit]
+                    id=Uri van Roe
+                    [filter_side]
+                        [allied_with]
+                            side=1
+                        [/allied_with]
+                    [/filter_side]
+                [/have_unit]
+            [/and]
+            [then]
+                {MSG_Uri _"...well, now that Gawen thinks he's won, there's no need to keep up this farce, is there, Bor?"}
+                [modify_side]
+                    side=4
+                    team_name=bad
+                [/modify_side]
+
+                {REPLACE_SCENARIO_MUSIC theme_of_a_new_order.ogg}
+                {MSG_Cryne _"Indeed, Uri, there isn't! Let our swords instead cut that underling's head off! And slay that witch, his step-mother, as well!"}
+                [modify_side]
+                    side=3
+                    team_name=bad
+                [/modify_side]
+                {APPEND_MUSIC battle/onethousandsuns.ogg}
+                {APPEND_MUSIC battle/battlecry.ogg}
+                {APPEND_MUSIC battle/gameplay06.ogg}
+                {APPEND_MUSIC siege_of_laurelmor.ogg}
+                {APPEND_MUSIC battle-epic.ogg}
+                # This macro (ANO4_TRAITORS) is defined in ano-01_06macros.cfg and contains most of the dialogue that occurs here:
+                {ANO4_TRAITORS _"Reach either the eastern or western signpost with both Lorin AND Reme. After that, hold on at least $ano_howmanyturns turns before the honorable death of Gawen Hagarthen."}
+            [/then]
+            [else]
+#ifdef DEBUG_MODE
+                {IF_DEBUG_MODE_IS_ACTUALLY_ON}
+                {MSG_narrator _"Using debug mode to kill all the enemies is cheating!"}
+                {ELSE}
+                {MSG_narrator _"DEBUG_MODE is defined; that means you probably cheated to kill all the enemies!"}
+                {END_IF}
+#else
+                {MSG_narrator _"Despite the incredible odds stacked against him, Gawen somehow managed to defeat all of his enemies at Barnon."}
+                {MSG_narrator _"However, this meant that he never managed to unite the warring factions of Wesnoth against the common foe that was due to invade..."}
```

Comment:
This message is a spoiler so I would definitely just keep it to "... unite the warring factions of Wesnoth..."

Reply:
Hmm that's fair. I guess I would say that is a spoiler too. We probably shouldn't spoil the player in the campaign description so that should get changed too at some point.

Reply:
(I copied the campaign description from `_main.cfg` into `readme.txt` too, btw, in 4b1edc3, so if we update the campaign description in the former, we should do it in the latter, too)